### PR TITLE
ReturnCount: enforce max=1 over checkstyle code (part 2)

### PR DESCRIPTION
**Pull request by Vladlis**
#3496

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java [hunk 1]
```diff
@@ -593,12 +593,13 @@ private String getNextImportGroup(int currentGroupNumber) {
      *        true, if current group contains at least one import.
      */
     private boolean hasAnyImportInCurrentGroup(String currentGroup) {
+        boolean result = false;
         for (ImportDetails currentImport : importToGroupList) {
             if (currentGroup.equals(currentImport.getImportGroup())) {
-                return true;
+                result = true;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: if we know the result, break out of the loop.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java [hunk 2]
```diff
@@ -123,11 +123,12 @@ public void visitToken(DetailAST ast) {
      * @return if {@code importText} contains an illegal package prefix
      */
     private boolean isIllegalImport(String importText) {
+        boolean isIllegal = false;
         for (String element : illegalPkgs) {
             if (importText.startsWith(element + ".")) {
-                return true;
+                isIllegal = true;
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java]: if we know the result, break out of the loop.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 3]
```diff
@@ -539,11 +539,12 @@ private static void verifyCheckstyleMessage(Map<String, List<String>> usedMessag
      */
     private static boolean isSubset(int[] array, int... arrayToCheckIn) {
         Arrays.sort(arrayToCheckIn);
+        boolean result = true;
         for (final int element : array) {
             if (Arrays.binarySearch(arrayToCheckIn, element) < 0) {
-                return false;
+                result = false;
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: same here.

**Comment by rnveach** [comment]
@Vladlis 2 tests are failing.
https://ci.appveyor.com/project/Checkstyle/checkstyle/build/7328/job/lixke9cl2bvf2mxv#L3100

File: config/suppressions.xml [hunk 4]
```diff
@@ -94,5 +94,5 @@
     <suppress id="returnCountMaxOne" files=".*[\\/]ant[\\/]|.*[\\/]filters[\\/]|.*[\\/]api[\\/]|.*[\\/]annotation[\\/]|
         |.*[\\/]coding[\\/]|.*[\\/]design[\\/]|.*[\\/]imports[\\/]|.*[\\/]indentation[\\/]|.*[\\/]javadoc[\\/]|
         |.*[\\/]naming[\\/]|.*[\\/]regexp[\\/]|.*[\\/]sizes[\\/]|.*[\\/]whitespace[\\/]|.*[\\/]doclets[\\/]|.*[\\/]gui[\\/]|
-        |.*[\\/]utils[\\/]|.*[\\/]doclets[\\/]|.*[\\/]internal[\\/]"/>
+        |.*[\\/]doclets[\\/]"/>
```
Comment (romani) [inline config/suppressions.xml]: we already have docklets, in line above

**Comment by romani** [comment]
@Vladlis , rnveach is right about regression testing, we recently did similar update simple update that resulted in exception and hot fix release.
Lets run affected Checks, and be sure that no regression and exception appeared.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java (hunk 1 above)
Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java (hunk 2 above)
Comment (Vladlis) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/IllegalImportCheck.java]: Done

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 3 above)
Comment (Vladlis) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: Done

File: config/suppressions.xml (hunk 4 above)
Comment (Vladlis) [inline config/suppressions.xml]: Fixed

**Comment by Vladlis** [comment]
Here are reports for all affected checks (**no regression**):
http://vladlis.github.io/3496/part2/1/
http://vladlis.github.io/3496/part2/2/
http://vladlis.github.io/3496/part2/3/

**Comment by rnveach** [comment]
> http://vladlis.github.io/3496/part2/3/

What is the contents of `import-control.xml`?
@romani Should external files like this be added to the diff reports? or should we use standard files?

**Comment by romani** [comment]
>Should external files like this be added to the diff reports?

Yes, it would be good to have. In this case we could trust Vladlis

**Comment by Vladlis** [comment]
@rnveach @romani 
I used Checkstyle config https://github.com/checkstyle/checkstyle/blob/master/config/import-control.xml

**Comment by romani** [comment]
@Vladlis , please rebase to check TC violations.

**Comment by Vladlis** [comment]
@romani
Done

**Pull request merged**